examples/CMakeLists.txt

```diff
@@ -46,17 +46,65 @@ IF(DEAL_II_COMPONENT_EXAMPLES)
     PATTERN "DTMB-5415_bulbous_bow.iges"  # step-54
     )
 
+  IF(DEAL_II_COMPILE_EXAMPLES)
+    #
+    # Set up all executables:
+    #
+    FILE(GLOB _steps ${CMAKE_CURRENT_SOURCE_DIR}/step-*/*.cc)
+    FOREACH(_step ${_steps})
+      GET_FILENAME_COMPONENT(_name ${_step} NAME_WE)
+      GET_FILENAME_COMPONENT(_directory ${_step} DIRECTORY)
+
+      #
+      # Extract dependency information from CMakeLists.txt file. We have
+      # to be careful with the negated logic used in the CMakeLists.txt
+      # file.
+      #
+      SET(_dependency_string TRUE)
+      IF(EXISTS "${_directory}/CMakeLists.txt")
+        FILE(STRINGS "${_directory}/CMakeLists.txt" _dependency_string
+          REGEX "^IF.*DEAL_II.* # keep in one line$"
+          )
+        STRING(REPLACE "IF(" "" _dependency_string "${_dependency_string}")
+        STRING(REPLACE ") # keep in one line" "" _dependency_string "${_dependency_string}")
+        IF("${_dependency_string}" STREQUAL "")
+          SET(_dependency_string "TRUE")
+        ELSE()
+          SET(_dependency_string "NOT (${_dependency_string})")
+        ENDIF()
+      ENDIF()
```

Comment:
That will work, but I'm a bit uneasy about the fragility of the approach -- you are relying on the cmake syntax used in the file, but I think we can only do this if we only test feature macros there (nothing else) and use the exact same syntax everywhere.

I still think that listing dependencies in some standard format somewhere in a separate file may be the better approach, but I'm not going to push for it. What do others think?